torch/_inductor/graph.py

```diff
@@ -320,6 +320,85 @@ def __init__(
         )
         self.init_backend_registration()
         self.module_storages_in_benchmark_use: Set[StorageDataPtr] = set()
+        self.tensors_for_benchmark_use: Dict[
+            torch.device, List[torch.Tensor]
+        ] = self.get_sorted_tensors_for_benchmark_use(gm)
+
+    def get_benchark_tensor(
+        self, needed_bytes: int, device: torch.device
+    ) -> Optional[torch.Tensor]:
+        sorted_tensors = self.tensors_for_benchmark_use[device]
+
+        def binary_search_needed_size():
+            left = 0
+            right = len(sorted_tensors) - 1
+            min_index = None
+
+            while left <= right:
+                mid = left + (right - left) // 2
+
+                if sorted_tensors[mid].untyped_storage().nbytes() > needed_bytes:
+                    min_index = mid
+                    right = mid - 1
+                else:
+                    left = mid + 1
+
+            return min_index
+
+        min_index = binary_search_needed_size()
+        if min_index is None:
+            return None
+
+        while (
+            min_index < len(sorted_tensors)
+            and sorted_tensors[min_index].untyped_storage().data_ptr()
+            in self.module_storages_in_benchmark_use
+        ):
+            min_index += 1
+
+        return sorted_tensors[min_index] if min_index < len(sorted_tensors) else None
+
+    def get_sorted_tensors_for_benchmark_use(
+        self, gm: torch.fx.GraphModule
+    ) -> Dict[torch.device, List[torch.Tensor]]:
+        seen_dp = set()
+        tensors_to_use = []
+
+        tc = torch._guards.TracingContext.try_get()
+        tc_params: List[torch.Tensor] = [] if tc is None else tc.params_flat  # type: ignore[assignment]
+
+        for mod_tensor in itertools.chain(tc_params, gm.parameters(), gm.buffers()):
+            mod_tensor = (
+                mod_tensor.data
+                if type(mod_tensor) is torch.nn.Parameter
+                else mod_tensor
+            )
+            if type(mod_tensor) is not torch.Tensor or not torch._C._has_storage(
+                mod_tensor
+            ):
+                continue
+
+            # MM benchmarking can be sensitive to mantissa
+            # Dont take from integer tensors bc they may be sparse
+            if not mod_tensor.is_floating_point():
+                continue
+
+            dp = mod_tensor.untyped_storage().data_ptr()
+            if dp in seen_dp:
+                continue
+
+            tensors_to_use.append(mod_tensor)
+            seen_dp.add(dp)
+
+        sorted_tensors = sorted(
+            tensors_to_use, key=lambda x: x.untyped_storage().nbytes()
+        )
+
+        sorted_by_device = defaultdict(list)
+        for t in sorted_tensors:
+            sorted_by_device[t.device].append(t)
```

Comment:
NIT: the `tensors_to_use` list isn't really needed, you can do everything inside the final dict.
```suggestion
            tensors_by_device[mod_tensor.device].append(mod_tensor)
            seen_dp.add(dp)

        for device_tensors in tensors_by_device.values():
            device_tensors.sort(key=lambda x: x.untyped_storage().nbytes())
```